compiler/src/dotty/tools/dotc/interactive/SourceTree.scala

```diff
@@ -10,7 +10,7 @@ import Contexts._, NameOps._, Symbols._
 import util._, util.Positions._
 
 /** A typechecked named `tree` coming from `source` */
-case class SourceTree(tree: tpd.NameTree, source: SourceFile) {
+case class SourceTree(topLevelImports: List[tpd.Import], tree: tpd.NameTree, source: SourceFile) {
```

Comment:
Shouldn't  an import be a source tree instead of attaching all top level import to every source tree?

Reply:
Yes I think this makes more sense, instead of storing topLevelImports + TypeDef of the class, we should just store the root PackageDef filtered to only contain the TypeDef corresponding to the class.